lldb/source/Expression/DWARFExpression.cpp

```diff
@@ -532,6 +532,73 @@ bool DWARFExpression::LinkThreadLocalStorage(
   return true;
 }
 
+/// Returns true if `opdcodes` contains the opcode for the register used for the
+/// Swift Async Context (x22 for aarch64, r14 for x86-64). It must also not
+/// contain any other opcodes.
+static bool IsAsyncRegCtxExpr(DataExtractor &opcodes,
+                              llvm::Triple::ArchType triple) {
+  offset_t offset = 0;
+  const uint8_t op_async_ctx_reg = opcodes.GetU8(&offset);
+  if (opcodes.BytesLeft(offset) != 0)
+    return false;
+
+  // FIXME: could this be exposed through the ABI/Language plugins?
+  return (triple == llvm::Triple::x86_64 && op_async_ctx_reg == DW_OP_reg14) ||
+         (triple == llvm::Triple::aarch64 && op_async_ctx_reg == DW_OP_reg22);
+}
+
+/// If `opcodes` contain the location of asynchronous contexts in Swift,
+/// evaluates DW_OP_call_frame_cfa and returns its result. Otherwise, does
+/// nothing. This is possible because, for async frames, the language unwinder
+/// treats the asynchronous context as the CFA of the frame.
+static llvm::Expected<Value> SwiftAsyncEvaluate_DW_OP_entry_value(
```

Comment:
Idea: In the future, languages could be able to register a DWARF expression plugin that matches sequences like this and returns a value?

Reply:
Yup, that could be an interesting idea, though I wonder how languages would use this...